include/deal2lkit/utilities.h

```diff
@@ -460,4 +460,27 @@ void smart_delete (SmartPointer<TYPE> &sp)
     }
 }
 
+/**
+ *  A simple class to shift a vector by a scalar.
+ *  This function is deprecated in deal but needed in many codes
+ */
+
+template <typename VEC>
+void vector_shift(VEC &in_vec, double a_scalar)
+{
+#ifdef DEAL_II_WITH_CXX11
+  for (auto i : in_vec.locally_owned_elements())
+    {
+      in_vec[i] += a_scalar;
+    }
+#else
+  for (unsigned int i = 0; i<in_vec.size(); ++i)
+    {
+      if (in_vec.locally_owned_elements().is_element(i))
+        in_vec[i] += a_scalar;
+    }
```

Comment:
with the version you wrote, the non cxx11 version has to loop over all elements, even if the processor owns only 1 element, and this, of course, is slower...